src/peergos/shared/user/fs/LazyInputStreamCombiner.java

```diff
@@ -68,22 +68,22 @@ public LazyInputStreamCombiner(WriterData version,
     public CompletableFuture<Boolean> getNextStream(int len) {
         return getSubsequentMetadata(this.nextChunkPointer, 0)
                 .thenCompose(access -> getChunk(access, len))
-                .thenApply(p -> {
-                    updateState(0,globalIndex + Chunk.MAX_SIZE, p.left, p.right);
+                .thenApply(t -> {
+                    updateState(0,globalIndex + Chunk.MAX_SIZE, t.left, t.middle, t.right);
                     return true;
                 });
     }
 
-    private CompletableFuture<Pair<byte[], AbsoluteCapability>> getChunk(CryptreeNode access, int truncateTo) {
+    private CompletableFuture<Triple<byte[], byte[], AbsoluteCapability>> getChunk(CryptreeNode access, int truncateTo) {
```

Comment:
Make a data-class rather  than a 3-tuple? Clearer to read than the ``left``, ``middle``, ``right`` dotted over the place